alibi_detect/cd/chisquare.py

```diff
@@ -2,15 +2,18 @@
 from scipy.stats import chi2_contingency
 from typing import Callable, Dict, List, Optional, Tuple, Union
 from alibi_detect.cd.base import BaseUnivariateDrift
+from alibi_detect.utils.warnings import deprecated_alias
 
 
 class ChiSquareDrift(BaseUnivariateDrift):
+    @deprecated_alias(preprocess_x_ref='preprocess_at_init')
     def __init__(
             self,
             x_ref: Union[np.ndarray, list],
             p_val: float = .05,
             categories_per_feature: Optional[Dict[int, int]] = None,
-            preprocess_x_ref: bool = True,
```

Comment:
This change has been done to draw a distinction between **whether or not** to preprocess x_ref (`x_ref_preprocessed`), and **when** to preprocess x_ref (`preprocess_at_init`). Original PR: https://github.com/SeldonIO/alibi-detect/pull/381

This is necessary since `save_detector` saves the preprocessed `x_ref` (if preprocessing performed at init), and so further preprocessing of `x_ref` should not be done when the detector is loaded again with `load_detector`.

The existing save/load functionality gets around this issue by saving two sets of kwargs within the `state_dict`; a set of kwargs to be used to instantiate the detector, and a set of kwargs to override attributes with post-instantiation. This approach isn't desirable now the config is human readable, hence the changes.

Reply:
Ah I see now this was merged into the current branch. Wonder if it's worth it splitting out and committing to master?

Reply:
I would say it makes sense and would separate concerns and reduce the size of this PR a bit!